src/bin/gen_all_data.rs

```diff
@@ -62,54 +60,72 @@ fn write_default_state_data() -> io::Result<()> {
         "done! wrote {} recommendations to {} in {:.2}s",
         N_RECOMMENDATIONS,
         at,
-        dur.as_secs_f64());
+        dur.as_secs_f64()
+    );
     Ok(())
 }
 
 fn write_ordered_allowed() -> io::Result<()> {
     let (dur, out) = timed(write_ordered_allowed_inner);
     let (name, lines) = out?;
-    eprintln!("done! wrote {} words to {} in {:.2}s", lines, name, dur.as_secs_f64());
+    eprintln!(
+        "done! wrote {} words to {} in {:.2}s",
+        lines,
+        name,
+        dur.as_secs_f64()
+    );
     Ok(())
 }
 
 fn write_ordered_allowed_inner() -> io::Result<(String, usize)> {
     let unordered = read_unordered_allowed_words()?;
     let ordered = read_ordered_frequency_data_words()?;
-    let to_write = ordered_words(&unordered, &ordered);
-
-    let at = format!("{}{}", EMBED_DATA_DIRECTORY, ORDERED_ALLOWED_WORDS_FILE_NAME);
-    let mut out = io::BufWriter::new(fs::File::options()
-        .truncate(true)
-        .create(true)
-        .write(true)
-        .open(&at)?);
+    let to_write = ordered_words(&unordered, &ordered).filter(|s| is_wordle_str(s));
+
+    let at = format!(
+        "{}{}",
+        EMBED_DATA_DIRECTORY, ORDERED_ALLOWED_WORDS_FILE_NAME
+    );
+    let mut out = io::BufWriter::new(
+        fs::File::options()
+            .truncate(true)
+            .create(true)
+            .write(true)
+            .open(&at)?,
+    );
 
     let mut count = 0;
     for item in to_write {
-        writeln!(out, "{}", item)?;
+        let compressed = CompressedWord::new(item);
+        assert_eq!(compressed.to_string(), item);
```

Comment:
we can probably change this to assert only in debug builds

Reply:
Since this is in gen_all_data, I don't know how often it would get built in debug mode, and I don't think performance of this part of gen_all_data matters much, but I don't feel strongly either way.